Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -511,6 +511,9 @@ void WebProcessPool::createGPUProcessConnection(WebProcessProxy& webProcessProxy
     parameters.presentingApplicationAuditToken = configuration().presentingApplicationProcessToken();
 #endif
 
+    parameters.useGPUProcessForDOMRenderingEnabled = WTF::anyOf(webProcessProxy.pages(), [](const auto& page) {
```

Comment:
I am a little worried we may create the GPUProcessConnection too eagerly and not have any pages yet. In this case, we set the flag to false and never update it. Shouldn't we send an updated flag to the GPUProcess whenever a page is added / removed from the WebProcess to be safe?

Reply:
That's a good point.